source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -456,10 +458,10 @@ StreamRateLimiter::StreamRateLimiter(uint64_t max_kbps, uint64_t max_buffered_da
       // ~63ms intervals.
       token_bucket_(SecondDivisor, time_source, SecondDivisor),
       token_timer_(dispatcher.createTimer([this] { onTokenTimer(); })),
-      buffer_(resume_data_cb, pause_data_cb) {
+      buffer_(resume_data_cb, pause_data_cb, overflow_data_cb) {
   ASSERT(bytes_per_time_slice_ > 0);
   ASSERT(max_buffered_data > 0);
-  buffer_.setWatermarks(max_buffered_data);
+  buffer_.setWatermarks(max_buffered_data / 2, max_buffered_data, max_buffered_data + 1);
```

Comment:
Interesting choice on OW limit. I know this has been one of the more discussed threads around OW, can you comment on how the limits that were set (or elided) in this PR were picked?

Reply:
I'll add more comments describing the reasons for the threshold values.
Specifically in this case, the buffer limit comes from my understanding of the following:
https://github.com/envoyproxy/envoy/blob/cfa29e187007331a93bfc83df5873aa7270b3980/include/envoy/http/filter.h#L682-L689

@avd suggested adding an external configuration. Do you think it would be beneficial to "increase" configuration complexity (add another field for overflow)?

Reply:
I think a runtime limit or scaling factor would be the right way to go here. I don't think most Envoy users want to be configuring this at a fine gain or via xDS.

Reply:
I'm not sure if all watermark buffers should have the same OW multiplier. However, I agree that it simplifies things.

Reply:
+1 to configuring in runtime and possibly just using a default multiplier unless the user explicitly calls another constructor? IMO we should try to simplify this for the common case as much as possible. I definitely don't think we should add actual config for this as I really can't see anyone setting it in the common case.